### DISCUSSION TITLE
Learning how to program ui design

### DISCUSSION THREAD

**Discussion by YouButWeaker** [ppixiv/ppixiv#91] [General]
What materials did you use or recommend for learning how to create an extension that changes the design and functions of a website?

**Comment by ppixiv** [comment]
I don't really know any in particular, but all you really need to know is general web development.  A user script is just JS, it's the same as scripting on a regular website.  I wouldn't start with this script, though, it's much more complex than most user scripts since it replaces the UI completely rather than just changing a few things (it's a full application that happens to have user script packaging).

**Reply by YouButWeaker** [reply]
yeah I was more curious about what it took to completely replace the UI and how much of it was general knowledge that could be carried over to a different website and how much of it was familiarity with pixiv's code

**Reply by ppixiv** [reply]
There's some specialized stuff for dealing with Pixiv (turning off their website so it doesn't keep doing stuff in the background is tricky and imperfect), but mostly it's just a regular web app, with some custom packaging to load it out of a single file so it can be loaded as a user script.  It doesn't use anything from Pixiv's website itself (only the API).  It's possible to run it as an independent local web app connecting to a local server (that's what VView is), but that's not currently packaged to make it easy to install.
